pkg/services/dashboards/service/dashboard_service.go

```diff
@@ -290,6 +290,78 @@ func (dr *DashboardServiceImpl) BuildSaveDashboardCommand(ctx context.Context, d
 	return cmd, nil
 }
 
+func (dr *DashboardServiceImpl) ValidateDashboardBeforeSave(ctx context.Context, dashboard *dashboards.Dashboard, overwrite bool) (bool, error) {
```

Comment:
FYI - Mainly copy and pasted from here: https://github.com/grafana/grafana/blob/322bd7839ece78b6f2d7c392890b0384187330f8/pkg/services/dashboards/database/database.go#L288